src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,1488 @@
+        //<editor-fold desc="ConjugatedPiSystemsDetector" defaultstate="collapsed">
+        Objects.requireNonNull(anAtomArray);
+        Objects.requireNonNull(aBondArray);
+        try {
+            IAtomContainerSet tmpConjugatedAtomContainerSet;
+            tmpConjugatedAtomContainerSet = ConjugatedPiSystemsDetector.detect(anAtomContainer);
+            //molecule mapping
+            //iterate over every atomcontainer from ConjPiSystemsDetector output
+            for (IAtomContainer tmpConjAtomContainer: tmpConjugatedAtomContainerSet.atomContainers()) {
+                for (IAtom tmpConjAtom: tmpConjAtomContainer.atoms()) {
+                    int tmpAtomInteger = tmpConjAtom.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_ATOM_INDEX_PROPERTY_KEY);
+                    IAtom tmpAtom = anAtomArray[tmpAtomInteger];
+                    tmpAtom.setProperty(AlkylStructureFragmenter.INTERNAL_ASF_CONJ_PI_MARKER_KEY, true);
+                }
+                for (IBond tmpConjBond: tmpConjAtomContainer.bonds()) {
+                    int tmpBondInteger = tmpConjBond.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_BOND_INDEX_PROPERTY_KEY);
+                    IBond tmpBond = aBondArray[tmpBondInteger];
+                    tmpBond.setProperty(AlkylStructureFragmenter.INTERNAL_ASF_CONJ_PI_MARKER_KEY, true);
+                }
+            }
+            aMolecularArraysInstance.setAtomArray(anAtomArray);
+            aMolecularArraysInstance.setBondArray(aBondArray);
+        } catch (Exception anException) {
+            AlkylStructureFragmenter.this.LOGGER.log(Level.WARNING, String.format(this.loggerExceptionStringFormat,
+                    anException.toString(), anAtomContainer.getProperty(Importer.MOLECULE_NAME_PROPERTY_KEY),
+                    "Conjugated Pi Systems detection failed."), anException);
+            throw new IllegalArgumentException(String.format(this.loggerExceptionStringFormat,
+                    anException.toString(), anAtomContainer.getProperty(Importer.MOLECULE_NAME_PROPERTY_KEY),
+                    "Conjugated Pi Systems detection failed."));
+        }
+
+        //</editor-fold>
+    }
+    /**
+     * Protected method to check given atomcontainer for disconnected structures.
+     *
+     * @param anAtomContainer IAtomContainer to check
+     * @return IAtomContainerSet containing partitioned structures as single IAtomContainer
+     */
+    protected IAtomContainerSet separateDisconnectedStructures(IAtomContainer anAtomContainer) throws IllegalArgumentException{
+        Objects.requireNonNull(anAtomContainer,"Given IAtomContainer is null.");
+        try {
+            IAtomContainerSet tmpFragmentSet = new AtomContainerSet();
+            //logger used only for debug purpose
+            AlkylStructureFragmenter.this.LOGGER.log(Level.INFO, System.currentTimeMillis() + " start sD, AC size: " + anAtomContainer.getAtomCount() + ", " + anAtomContainer.getBondCount());
```

Comment:
Most of what you are logging as info is actually a debug message. It is clogging up the MORTAR log file! Please change that.